### ISSUE TITLE
cannot set blank username and password

### ISSUE THREAD

**Issue by beniquezsd** [Jermolene/TiddlyWiki5#2682]
Hello, please bear with me, I've been trying to write this out as clearly as possible. 

### Some Background
I've been trying to launch TiddlyWiki5 on my webhost, which uses AFS for its filesystem.

In order to run unattended commands, I've written a script (`unattended.sh`) that uses k5start to grab a kerberos token and launch the TiddlyWiki sever with the correct ticket, so the app does not lose access to the filesystem after my shell token expires.

I also launch separate web applications and multiple sites with other "launcher" scripts, most of which look like the following:
```bash
unattended.sh  ${SESSION_NAME} "${NODE}  mywiki --server ${NODE_PORT} $:/core/save/all text/plain text/html '' ''  127.0.0.1 /wiki"
```

### The Problem
Whenever I launch TiddlyWiki in this manner, **no matter how I escape the quotes,** or wrap the entire command in quotations, **the wiki thinks it has a set username and password** (due to how Bash interprets whitespace).

### Solutions I've tried
I've tried everything from escaping inner double-quotes, using single-quotes, removing the outer quotes, etc... all  to no avail. When I go to the wiki's URL, I get a login screen, that _sometimes_ accepts "" and "" as the user name and password.

#### What finally worked
I've gotten around the issue entirely by hard-coding the parameter to `null` in `core/modules/commands/server.js` and adjusting the code to shift the parameters. 

### TL;DR:
With the current TiddlyWiki authentication scheme...
* Having optional positional parameters in the middle of a command string can cause headaches when dealing with shells, parameter passing, and white space, as noted above.
* On most shared hosting, anyone on the system can execute a simple  `ps -aux | grep tiddlywiki` and get the  the credentials for the wiki.

#### Solutions:
As far as I see it...
1. One way around this would be to make sure the username and password fields (as well as the other optional fields) are always the last two fields, allowing users to easily omit the parameters entirely. But this might backfire if they don't want to specify any of the other optional parameters.
1. Use named paremeters. ( This will increase code complexity)
1. Replace the username and password fields with a "credentials-file" field, the contents of which can be left blank (or specified as `/dev/null`) to indicate a no access control.
1. Use a Tiddler to store the authentication data. 

I don't know the codebase well enough to know if TiddlyWiki can read Tiddlers from the context of the `core/modules/commands/server.js` file, so I'm leaning more towards solution no. 3, as listed above. I don't need it to be super secure, but it would be nice if it wasn't easily accessible on the command-line, either.

I wanna give the other developers a chance to give their opinion and maybe see if better ideas come up.
I'm willing to submit a PR if this is fine by everyone else. Any thoughts?

**Comment by Jermolene** [comment]
Hi @beniquezsd apologies for the delayed response.

I don't quite understand what's going on. The intention is that passing a blank username and password will skip the credentials check:

https://github.com/Jermolene/TiddlyWiki5/blob/master/core/modules/commands/server.js#L106

Can you report on the values of the username and password variables at that point in your setup?

I'm not at all knowledgeable about Bash; the script that I use for starting my local dev server has many improvements from @pmario and others, but it does appear to have the property of allowing the `--server` command to be issued with the username and password omitted. I generally start it with a simple `./bin/serve.sh`, and in that default form I am able to access the wiki without a username and password:

https://github.com/Jermolene/TiddlyWiki5/blob/master/bin/serve.sh#L72

**Comment by beniquezsd** [comment]
I'm sorry if I made it a bit confusing, but what's going on for me, is that I'm trying to *disable* the username and password.

Due to how my shell interprets strings, nodejs is given literally "" for the username and literally "" for the password,  which is not what I intended...

I also think if users want to password protect their wiki, they wouldn't want other users on shared hosting to see the credentials on the process command line. I understand the kernel could be configured to hide that, but TiddlyWiki users might not have that level of control when it comes to shared hosting.

**Comment by beniquezsd** [comment]
(Excuse me for not editing, I'm on mobile) 

Since I want a wiki in a subdirectory, I cant just drop the username and password fields. I NEED to specify they're blank, but the shell weirdness isn't allowing me to do so.

**Comment by pmario** [comment]
The --listen command is able to deal with that. --server is deprecated.
